### PR TITLE
Transfers: Add optional srm https compatibility Fix #4944

### LINKED ISSUE #4944: SRM to HTTPS optional compatibility
Motivation
----------
https://github.com/rucio/rucio/blob/882b662a24c75d515080606be635959566b17f14/lib/rucio/common/constants.py#L42
Defines a scheme_map between protocol compatibility in Rucio. We need an additional compatibility between `srm` and `https` (and vice-versa) which is only optionally (based on a config parameter) activated.

Modification
------------
Check if an option
```
[transfers]
srm_https_compatibility = True
```
is set and add it to the scheme_map.

### PR DESCRIPTION
**Pull request by joeldierkes**

### CODE REVIEW

**Comment by joeldierkes** [comment]
`etc/rucio.cfg.template` had the execution flag set on the file, I reverted that (`chmod -x etc/rucio.cfg.template`).

File: etc/rucio.cfg.template [hunk 1]
```diff
@@ -253,3 +253,6 @@ admin_issuer = wlcg
 
 [api]
 #endpoints = accountlimits, accounts, config, credentials, dids, export, heartbeats, identities, import, lifetime_exceptions, locks, meta, ping, redirect, replicas, requests, rses, rules, scopes, subscriptions
+
+[transfers]
```
Comment (mlassnig) [inline etc/rucio.cfg.template:259]: maybe part of [conveyor]?

Comment (joeldierkes) [reply etc/rucio.cfg.template:259]: The issue explicitly stated that it should be under `[transfer]`. `[conveyor]` makes sense to me. What do you think @bari12 ?

Comment (bari12) [reply etc/rucio.cfg.template:259]: Should be transfers, since we want to get away from the code-names anyway.

**Comment by bari12** [comment]
Rebase please; otherwise gtg

**Review by bari12 — APPROVED** [review]

**Pull request merged**